include/openPMD/RecordComponent.hpp

```diff
@@ -29,10 +29,41 @@
 #include <queue>
 #include <string>
 #include <stdexcept>
+#include <vector>
+#include <array>
 
 
 namespace openPMD
 {
+//! @{
+/** Share ownership with a raw pointer
+ *
+ * Helper function to share write ownership
+ * unprotected and without reference counting with a
+ * raw pointer or stdlib container.
+ */
```

Comment:
add `@note` or `@warning` about user-responsibility of memory pre-allocation, buffer-consistency, etc.